src/components/experiment-tracking/run-export-modal.js/run-export-modal.js

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useCallback, useContext, useEffect } from 'react';
+import { CSVLink } from 'react-csv';
+
+import { constructExportData } from '../../../utils/experiment-tracking-utils';
+import { ButtonTimeoutContext } from '../../../utils/button-timeout-context';
+
+import Button from '../../ui/button';
+import Modal from '../../ui/modal';
+
+import './run-export-modal.css';
+
+const RunExportModal = ({
+  theme,
+  visible,
+  setShowRunExportModal,
+  runMetadata,
+  runTrackingData,
+}) => {
+  const [exportData, setExportData] = useState([]);
+  const { isSuccessful, showModal, handleClick } =
+    useContext(ButtonTimeoutContext);
+
+  const updateExportData = useCallback(() => {
+    setExportData(constructExportData(runMetadata, runTrackingData));
+  }, [runMetadata, runTrackingData]);
+
+  // only if the component is visible first, then apply isSuccessful to show or hide modal
+  useEffect(() => {
+    if (visible && isSuccessful) {
+      setShowRunExportModal(showModal);
+    }
+  }, [showModal, setShowRunExportModal, isSuccessful, visible]);
+
+  return (
+    <div className="pipeline-run-export-modal pipeline-run-export-modal--experiment-tracking">
+      <Modal
+        closeModal={() => setShowRunExportModal(false)}
+        theme={theme}
+        title="Export experiment run"
+        visible={visible}
+      >
+        <div className="run-export-modal-button-wrapper">
+          <Button mode="secondary" onClick={() => setShowRunExportModal(false)}>
+            Cancel
+          </Button>
+          <CSVLink
+            data={exportData}
+            asyncOnClick={true}
+            onClick={updateExportData}
+            filename="run-data.csv"
+          >
+            <Button
+              onClick={handleClick}
+              mode={isSuccessful ? 'success' : 'primary'}
+            >
```

Comment:
Referencing the design, I think we want the `small` size button here. There's a `size` prop you can use.

Reply:
done, thank you